app/models/time_sheet.rb

```diff
@@ -19,27 +19,39 @@
 #  last_modified         :date
 #  status                :integer          default(0)
 #  signer                :integer
+#  wimi_signed           :boolean          default(FALSE)
+#  hand_in_date          :date
 #
 
 class TimeSheet < ActiveRecord::Base
-  belongs_to :user
-  belongs_to :project
+    belongs_to :user
+    belongs_to :project
+    enum status: [ :pending, :accepted, :rejected]
 
-  validates :workload_is_per_month, inclusion: {in: [true, false]}
-  validates :salary_is_per_month, inclusion: {in: [true, false]}
+    validates :workload_is_per_month, inclusion: { in: [true, false] }
+    validates :salary_is_per_month, inclusion: { in: [true, false] }
 
-  def self.time_sheet_for(year, month, project, user)
-    if project.nil?
-      return nil
-    else
-      sheets = TimeSheet.where(year: year, month: month, project: project, user: user)
-      if sheets.empty?
-        return create_new_time_sheet(year, month, project, user)
+    def sum_hours
+      hour_counter = 0
+      monthyearrange = Date.new(self.year, self.month, 1)..Date.new(self.year, self.month, Time.days_in_month(self.month, self.year))
+      WorkDay.where(date: monthyearrange, project: self.project_id, user_id: self.user_id).each do |day| 
```

Comment:
...project: project, user: user) sollte auch funktionieren :)
In Zeile 36 brauchst du die self's auch nicht